### PR TITLE
feat(collator-rewards): modifying the pallet

### PR DESCRIPTION
**Pull request by niall-watr**

### CODE REVIEW

File: pallets/collator-selection/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,567 @@
...
+			// `reward` is half of pot account minus ED, this should never fail.
+			let _success = T::Currency::transfer(&fee_pot, &author, fee_reward, KeepAlive);
+			let collator_reward = Self::collator_reward();
+			let _collator_success: DispatchResult = match T::Currency::free_balance(&collator_pot)
+				.checked_sub(&T::Currency::minimum_balance().add(collator_reward))
+			{
+				Some(_) =>
+					T::Currency::transfer(&collator_pot, &author, collator_reward, KeepAlive),
+				None => T::Currency::transfer(&collator_pot, &author, Zero::zero(), KeepAlive),
+			};
```
Comment (peterwht) [inline pallets/collator-selection/src/lib.rs:522]: Something like `let _collator_success = T::Currency::transfer(&collator_pot, &author, collator_reward, KeepAlive)` should be sufficient when using the `KeepAlive` flag.

File: pallets/collator-selection/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,567 @@
...
+		/// The currency mechanism.
+		type Currency: ReservableCurrency<Self::AccountId>;
+
+		/// Origin that can dictate updating parameters of this pallet.
+		type UpdateOrigin: EnsureOrigin<Self::RuntimeOrigin>;
+
+		/// Account Identifier from which the internal Pot is generated.
+		type PotId: Get<PalletId>;
+
+		/// Acount Identifier for Collator Rewards Account
```
Comment (peterwht) [inline pallets/collator-selection/src/lib.rs:125]:
```suggestion
		/// Account Identifier for Collator Rewards Account
```

File: pallets/collator-selection/src/lib.rs (hunk 1 above)
Comment (niall-watr) [reply pallets/collator-selection/src/lib.rs:522]: I was more concerned with doing a balance check to ensure that balance > reward, if we can get away with removing this check, then all the better

Comment (peterwht) [reply pallets/collator-selection/src/lib.rs:522]: Fair enough. The balances pallet will also ensure the reward will not be transferred unless the balance is high enough. There's no harm in explicitly checking, though! LGTM

**Comment by niall-watr** [comment]
Closed -> using astar block reward pallet as an alternative approach

**Pull request closed without merging**